nvtabular/dispatch.py

```diff
@@ -177,14 +177,34 @@ def _read_dispatch(df: DataFrameType = None, cpu=None, collection=False, fmt="pa
     return getattr(_mod, _attr)
 
 
-def _parquet_writer_dispatch(df: DataFrameType):
+def _parquet_writer_dispatch(df: DataFrameType, path=None, **kwargs):
     """Return the necessary ParquetWriter class to write
     data of a specified type.
+
+    If `path` is specified, an initialized `ParquetWriter`
+    object will be returned.  To do this, the pyarrow schema
+    will be inferred from df, and kwargs will be used for the
+    ParquetWriter-initialization call.
     """
+    _args = []
+    if isinstance(df, pd.DataFrame):
+        _cls = pq.ParquetWriter
+        if path:
+            _args.append(pa.Table.from_pandas(df, preserve_index=False).schema)
+    else:
+        _cls = cudf.io.parquet.ParquetWriter
+
+    if not path:
+        return _cls
+    return _cls(path, *_args, **kwargs)
```

Comment:
I'm wondering if we can get rid of the _write_table_dispatch call entirely, and return an object where write_table works consistently between cpu/gpu versions.

We could either have an adaptor class that handles the conversion from pandas to pyarrow, or we could even just monkey patch the write_table call for the cpu version like this:
```suggestion
    ret = _cls(path, *_args, **kwargs)
    if isinstance(df, pd.DataFrame):
        ret.write_table = lambda df: _cls.write_table(ret, pa.Table.from_pandas(df, preserve_index=False))
    return ret
```

Reply:
Sounds good to me :)